src/mdshow.jl

```diff
@@ -93,91 +69,82 @@ function Base.show(io::IO, ::MIME"text/markdown", m::MixedModel)
         bname = Symbol(bname)
 
         for (j, sig) in enumerate(m.σs)
             if bname in keys(sig)
-                print(io, "$(lpad(aligncompact(getproperty(sig, bname), σwidth),σcolwidth))")
+                push!(newrow, Ryu.writefixed(getproperty(sig, bname),digits))
             else
-                print(io, " "^σcolwidth)
+                push!(newrow, " ")
             end
-            print(io, "|")
         end
-        println(io)
+        push!(rows, newrow)
     end
 
-    dispersion_parameter(m) && println(io, "|$(rpad(_dname(m), bnwidth))|$(string(dispersion(m))[1:cowidth])||||$("|"^nrecols)")
+    if dispersion_parameter(m)
+        newrow = [_dname(m), Ryu.writefixed(dispersion(m),digits), "", "", ""]
+        for rr in fnames(m)
+            push!(newrow, "")
+        end
+        push!(rows, newrow)
+    end
 
-    return nothing
+    tbl = Markdown.Table(rows, align)
+    show(io, Markdown.MD(tbl))
 end
 
 
 function Base.show(io::IO, ::MIME"text/markdown", s::OptSummary)
-    println(io,"| | |")
-    println(io,"|-|-|")
-    println(io,"|**Initialization**| |")
-    println(io,"|Initial parameter vector|", s.initial,"|")
-    println(io,"|Initial objective value|", s.finitial,"|")
-    println(io,"|**Optimizer settings**| |")
-    println(io,"|Optimizer (from NLopt)|`", s.optimizer,"`|")
-    println(io,"|`Lower bounds`|", s.lowerbd,"|")
-    println(io,"|`ftol_rel`|", s.ftol_rel,"|")
-    println(io,"|`ftol_abs`|", s.ftol_abs,"|")
-    println(io,"|`xtol_rel`|", s.xtol_rel,"|")
-    println(io,"|`xtol_abs`|", s.xtol_abs,"|")
-    println(io,"|`initial_step`|", s.initial_step,"|")
-    println(io,"|`maxfeval`|", s.maxfeval,"|")
-    println(io,"|**Result**| |")
-    println(io,"|Function evaluations|", s.feval,"|")
-    println(io,"|Final parameter vector|", round.(s.final; digits=4),"|")
-    println(io,"|Final objective value|", round.(s.fmin; digits=4),"|")
-    println(io,"|Return code|`", s.returnvalue,"`|")
+    rows = [["",""]]
+
+    push!(rows, ["**Initialization**", ""])
+    push!(rows, ["Initial parameter vector", string(s.initial)])
+    push!(rows, ["Initial objective value", string(s.finitial)])
+
+    push!(rows, ["**Optimizer settings** ", ""])
+    push!(rows, ["Optimizer (from NLopt)", "`$(s.optimizer)`"])
+    push!(rows, ["`Lower bounds`", string(s.lowerbd)])
+    push!(rows, ["`ftol_rel`", string(s.ftol_rel)])
+    push!(rows, ["`ftol_abs`", string(s.ftol_abs)])
+    push!(rows, ["`xtol_rel`", string(s.xtol_rel)])
+    push!(rows, ["`xtol_abs`", string(s.xtol_abs)])
+    push!(rows, ["`initial_step`", string(s.initial_step)])
+    push!(rows, ["`maxfeval`", string(s.maxfeval)])
+
+    push!(rows, ["**Result**",""])
+    push!(rows, ["Function evaluations", string(s.feval)])
+    push!(rows, ["Final parameter vector", "$(round.(s.final; digits=4))"])
+    push!(rows, ["Final objective value", "$(round.(s.fmin; digits=4))"])
+    push!(rows, ["Return code", "`$(s.returnvalue)`"])
```

Comment:
```suggestion
    rows = [["", ""],

            ["**Initialization**", ""],
            ["Initial parameter vector", string(s.initial)],
            ["Initial objective value", string(s.finitial)],

            ["**Optimizer settings** ", ""],
            ["Optimizer (from NLopt)", "`$(s.optimizer)`"],
            ["`Lower bounds`", string(s.lowerbd)],
            ["`ftol_rel`", string(s.ftol_rel)],
            ["`ftol_abs`", string(s.ftol_abs)],
            ["`xtol_rel`", string(s.xtol_rel)],
            ["`xtol_abs`", string(s.xtol_abs)],
            ["`initial_step`", string(s.initial_step)],
            ["`maxfeval`", string(s.maxfeval)],

            ["**Result**",""],
            ["Function evaluations", string(s.feval)],
            ["Final parameter vector", "$(round.(s.final; digits=4))"],
            ["Final objective value", "$(round.(s.fmin; digits=4))"],
            ["Return code", "`$(s.returnvalue)`"]]
```
This is how I'd do it, FWIW, feel free to take it or leave it

Reply:
I was wondering whether you would catch some of the inefficiencies in all the `push!`ing, but a lot of it was just a convenient way to match the REPL-stuff that used `print` at each of those points.

Reply:
I figured that was the case